src/main/kotlin/org/rust/ide/refactoring/inlineFunction/RsInlineFunctionProcessor.kt

```diff
@@ -226,6 +229,32 @@ class RsInlineFunctionProcessor(
         }
     }
 
+    private fun removeFunctionUseStatements(caller: PsiElement, reference: RsFunction) {
+        usagesAsReference.filter {
+            it.element.ancestorOrSelf<RsUseSpeck>() != null }.forEach {
+            val useGroup = it.element.ancestorOrSelf<RsUseGroup>()
+            val useGroupChildrenSize = useGroup?.useSpeckList?.size
+
+            if (useGroupChildrenSize != null && useGroupChildrenSize > 1) {
```

Comment:
Just checking the nearest group is not enough, there may be other use groups above it and in that case they would be deleted. Consider something like this:
```rust
use foo::{bar::{test}, bar::test2};
fn main() {
    test();
}
mod foo {
    mod bar {
        pub fn /*caret*/test() {
            println!("test");
        }
        pub fn test2() {
            println!("test2");
        }
    }
}
```
It will remove too many imports. You should also probably check if the group is a sole child of a use item before deleting the whole use item.

Reply:
Fixed. Quoting the extended solution from my commit message:

> Added handling of multiple levels of use groups.
> 
> Now stopping on the first one that has multiple uses inside it, and deleting from it. Only if all groups have a single use ?> statement inside it will delete the whole use statement.